keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1157 @@
+
+#### Beta
+
+- The implementation works well with the Ensure secret pull images KEP and supports pod image pull policy set to `Always`
+- Add metrics
+
+#### GA
+
+- Gather feedback
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+This feature is feature gated so explicit opt-in is required on upgrade and explicit opt-out is required on downgrade.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+To migrate to the new approach,
+
+1. KAS will need to be updated to the new version to configure the list of audiences for which the kubelet is allowed to generate service account tokens for image pulls via the `--allowed-kubelet-audiences` flag.
+2. kubelet on the node needs to be updated to enable the feature flag and configure the credential provider to use service account tokens for image pull via the `TokenAttributes` field in the kubelet credential provider configuration.
+   1. The credential provider plugin will need to be updated to use the service account token for the audience configured in the kubelet credential provider configuration.
+
+Migration of the workloads to the new approach can be done per image or per registry basis by configuring the credential provider multiple times with different names for the same plugin (w or w/o service account tokens, different audiences).
+
+When things can fail:
+
+1. If the kubelet is updated to enable the feature flag and the credential provider is configured with the `TokenAttributes` field set, but the KAS is not updated with `--allowed-kubelet-audiences` to allow the kubelet to generate service account tokens for the audience configured in the kubelet credential provider configuration, the image pull will fail. The old KAS will reject the token request from the kubelet.
```

Comment:
> The old KAS will reject the token request from the kubelet.

It's important that this will actually fail instead of being allowed for the safety of this feature on downgrade and disable, right?  Will you manually test this to be sure?  I'm concerned because I don't remember this failing closed today.

Reply:
> We can make is so that the audience validation is beta/enabled by default at least one release before the rest of the functionality reaches beta/enabled by default. WDYT?

That works for me.  Let's get this clarified in the KEP beta criteria please so we don't forget.  I think that means two gates.

Reply:
Added a new feature gate and updated the beta criteria to include what @enj mentioned.